### PR TITLE
Modernize the embedded Puppet module

### PR DESCRIPTION
**Pull request by ekohl**
This modernizes the Puppet module to use the new functions API. This avoids a deprecation warning in Puppet 6.

It doesn't update the progress to new API since the monkey patching is broken in Puppet 6. To avoid this, it adds a version check in the main class so we don't include it.

To ensure it works as expected, we add module testing. This is not yet integrated into CI.

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like there's a bundler issue in more modern rubies. Need to investigate that.

**Comment by ekohl** [comment]
:green_heart:

File: modules/kafo_configure/lib/puppet/functions/kafo_configure/to_yaml.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Puppet::Functions.create_function(:'kafo_configure::to_yaml') do
+  dispatch :to_yaml do
+    param 'Hash', :variables
+    param 'Hash', :lookups
+    return_type 'String'
+  end
+
+  def to_yaml(variables, lookups)
+    dump = variables.merge(lookups)
+    YAML.dump(dump) + "\n...\n"
```
[the remark this answers is not in the document]
Comment (ekohl) [reply modules/kafo_configure/lib/puppet/functions/kafo_configure/to_yaml.rb]: [YAML document markers](https://yaml.org/spec/1.2/spec.html#id2800401) start with `---` (which you see a lot). You can mark the end with `...` and I think this is how we ensure there are no warnings included or other output when we parse the puppet output.

**Comment by ekohl** [comment]
> I didn't see anything suspicious and it looks good. Do you prefer to squash or keep commits separate? Do you want to add something more?

I'm happy with separate commits and do not intend to add more commits to this PR.

**Comment by ares** [comment]
Ok, I have tested with puppet 5.5.10 on Foreman nightly installer and all works, I suppose you were testing this with Puppet 6. I'm merging now, thanks @ekohl, please let me know if you want me to release a new version or you plan to add something else soon. This does not seem to be dropping anything so 2.2.0 is will be appropriate.

**Comment by ekohl** [comment]
I'd like to go over the current open PRs and see what else can be done before releasing. Probably not a lot.

**Pull request merged**